clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -123,16 +121,48 @@ export class HttpConnection implements IConnection {
         }
     }
 
-    private createTransport(requestedTransport: TransportType | ITransport, availableTransports: IAvailableTransport[], requestedTransferFormat: TransferFormat): ITransport {
+    private async setNegotiationResponse(headers: any): Promise<void> {
+        const response =  await this.httpClient.post(this.resolveNegotiateUrl(this.baseUrl), {
+            content: "",
+            headers,
+        });
+        this.negotiationResponse = JSON.parse(response.content as string);
+        this.connectionId = this.negotiationResponse.connectionId;
+        this.url = this.baseUrl + (this.baseUrl.indexOf("?") === -1 ? "?" : "&") + `id=${this.connectionId}`;
+        return Promise.resolve();
+    }
+
+    private async createTransport(requestedTransport: TransportType | ITransport, requestedTransferFormat: TransferFormat): Promise<void> {
         if (this.isITransport(requestedTransport)) {
             this.logger.log(LogLevel.Trace, "Connection was provided an instance of ITransport, using that directly.");
-            return requestedTransport;
+            this.transport = requestedTransport;
+            this.transport.onreceive = this.onreceive;
+            this.transport.onclose = (e) => this.stopConnection(true, e);
+            await this.transport.connect(this.url, requestedTransferFormat, this);
+
+            // only change the state if we were connecting to not overwrite
+            // the state if the connection is already marked as Disconnected
+            this.changeState(ConnectionState.Connecting, ConnectionState.Connected);
+            return Promise.resolve();
         }
 
-        for (const endpoint of availableTransports) {
+        const transports = this.negotiationResponse.availableTransports;
+        for (const endpoint of transports) {
             const transport = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
             if (typeof transport === "number") {
-                return this.constructTransport(transport);
+                this.transport = this.constructTransport(transport);
+                if (this.negotiationResponse.connectionId == null) {
```

Comment:
`===` or face the wrath of the linter gods.

Reply:
`===` FOR THE EQUALITY GODS

Reply:
@mikaelm12 _All animals are equal, but some animals are more equal than others_